web/src/app/services/data-store/data-store.service.ts

```diff
@@ -249,27 +252,29 @@ export class DataStoreService {
       .set(jobToDocument(job));
   }
 
-  async deleteSurvey(survey: Survey) {
+  async deleteSurvey(survey: Survey): Promise<void> {
```

Comment:
Can we do this in a single atomic transaction? Otherwise if the process gets interrupted the db will end up in an invalid state.

https://firebase.google.com/docs/firestore/manage-data/transactions

Reply:
done